pkg/activator/util/header.go

```diff
@@ -24,6 +24,7 @@ import (
 )
 
 var headersToRemove = []string{
+	activator.RequestCountHTTPHeader,
```

Comment:
Would the underlying issue be fixed if we removed the `ResponseCountHTTPHeader` here? Are two distinct headers needed in that case?

Reply:
The ResponseCountHTTPHeader header is set after the queue proxy returns to the activator, thus there is nothing to remove in the queue-proxy.

Reply:
FWIW I did play around with removing the response header so we'd only need one header in the activator again.

```go
origResponseModifier := p.ModifyResponse
p.ModifyResponse = func(r *http.Response) error {
	for _, h := range responseHeadersToRemove {
		r.Header.Del(h)
	}

	if origResponseModifier != nil {
		return origResponseModifier(r)
	}
	return nil
}
```

In belows `SetupHeaderPruning` function seemed to work just fine in unit and manual tests with the activator. If we want this PR in quickly, we don't need to block on this and can fix later. It's a worthwhile thing doing though ultimately I think.

Reply:
I tried essentials the same code and it didn't work for me. Maybe, I didn't get the change properly deployed. I'll try again.